metecho/api/model_mixins.py

```diff
@@ -162,3 +164,41 @@ def finalize_create_pr(self, *, error=None, originating_user_id):
             )
         else:
             self.notify_error(error=error, originating_user_id=originating_user_id)
+
+
+class SoftDeleteQuerySet(models.QuerySet):
+    def active(self):
+        return self.filter(deleted_at__isnull=True)
+
+    def delete(self):
+        soft_delete_child_class = getattr(self.model, "soft_delete_child_class", None)
+        if soft_delete_child_class:
+            parent = camel_to_snake(self.model.__name__)
+            soft_delete_child_class(None).objects.filter(
+                **{f"{parent}__in": self}
+            ).delete()
+        return self.active().update(deleted_at=timezone.now())
+
+    delete.queryset_only = True
+
+
+class SoftDeleteMixin(models.Model):
+    class Meta:
+        abstract = True
+
+    deleted_at = models.DateTimeField(null=True, blank=True)
+
+    objects = SoftDeleteQuerySet.as_manager()
+
+    def delete(self, *args, **kwargs):
+        soft_delete_child_class = getattr(self, "soft_delete_child_class", None)
+        if soft_delete_child_class:
+            parent = camel_to_snake(self.__class__.__name__)
+            soft_delete_child_class().objects.filter(**{parent: self}).delete()
```

Comment:
@wlonk When ScratchOrgs are deleted through this cascade, they are not being deleted on SF as well. I think this would need to call `queue_delete` instead?

Reply:
I added special-casing that should handle that.